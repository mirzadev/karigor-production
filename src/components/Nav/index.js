import React from "react";
import karigorLogo from "./images/karigor-logo.PNG";
import { capitalizeFirstLetter } from "../../utils/helper";

function Nav() {
  const categories = [
    {
      name: "Cultural",
      description:
        "Cultural activities of Karigor in various years. Major event for Karigor is annual Bengali New year celebration.",
    },
    {
      name: "Children Section",
      description:
        "Today's childrens are our future, who will uphold the Bengali culture in future in USA",
    },
    { name: "Education", description: "Delicious delicacies" },
    {
      name: "Social Activities",
      description:
        "Besides organizing annual cultural activities and events, Karigor is comitted to assist organizing social activities of other Bangladeshi organizarions in USA.",
    },
  ];

  const handleClick = () => {
    console.log("click handled");
  };

  return (
    <header data-testid="header" className="flex-row px-1">
      <h2>
        <a className="navbar-brand" id="brandName" href="/">
          <span>
            <img
              id="karigor-logo"
              src={karigorLogo}
              alt=""
              height="70"
              width="110"
            ></img>
          </span>{" "}
        </a>
      </h2>
      <a className="title-header">
        <span>KARIGOR PRODUCTION</span>
      </a>
      <nav className="right-aligned">
        <ul className="flex-row" id="nav-items">
          <li className="mx-2">
            <a href="#home">Home</a>
          </li>
          <li className="mx-2">
            <a href="#about">About</a>
          </li>
          <li className="mx-2">
            <a href="#contact">Contact Us</a>
          </li>
          <li className="mx-2">
            <a href="#events">Events</a>
          </li>
          {categories.map((category) => (
            <li className="mx-1" key={category.name}>
              <span
                onClick={() => {
                  handleClick();
                }}
              >
                {capitalizeFirstLetter(category.name)}
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

/*
function Nav() {
  const categories = [
    {
      name: "commercial",
      description:
        "Photos of Pohela Boishakh, cultural events, social activities",
    },
    {
      name: "children",
      description:
        "Childrens of Karigor regularly gets training on Bangali culture and regularly takes part in social activities",
    },
    {
      name: "religion",
      description: "Karigor upholds the religious and ethnic values for all",
    },
    {
      name: "mix-pictures",
      description: "Pictures of various events",
    },
  ];
  function categorySelected(name) {
    console.log(`${name} clicked`);
  }
  return (
    <header>
      <h2>
        <a className="navbar-brand" id="brandName" href="/">
          <span>
            <img
              id="karigor-logo"
              src={karigorLogo}
              alt=""
              height="60"
              width="100"
            ></img>
          </span>{" "}
          KARIGOR
        </a>
      </h2>
      <nav>
        <ul className="flex-row">
          <li className="mx-2">
            <a href="#home">Home</a>
          </li>
          <li className="mx-2">
            <a href="#about">About</a>
          </li>
          <li className="mx-2">
            <a href="#contact">Contact Us</a>
          </li>
          <li className="mx-2">
            <a href="#events">Events</a>
          </li>
          {categories.map((category) => {
            <li className="mx-1" key={category.name}>
              <span onClick={() => categorySelected(category.name)}>
                {category.name}
              </span>
            </li>;
          })}
        </ul>
      </nav>
    </header>
  );
}
*/

export default Nav;
