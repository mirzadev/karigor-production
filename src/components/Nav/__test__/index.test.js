import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "../../Nav";

afterEach(cleanup);

describe("Nav component", () => {
  //baseline tset

  it("renders", () => {
    render(<Nav />);
  });

  //snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(<Nav />);
    //assert value comparison
    expect(asFragment()).toMatchSnapshot();
  });
});
