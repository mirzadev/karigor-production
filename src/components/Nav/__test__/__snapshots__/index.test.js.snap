// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
    data-testid="header"
  >
    <h2>
      <a
        class="navbar-brand"
        href="/"
        id="brandName"
      >
        <span>
          <img
            alt=""
            height="70"
            id="karigor-logo"
            src="karigor-logo.PNG"
            width="110"
          />
        </span>
         
      </a>
    </h2>
    <a
      class="title-header"
    >
      <span>
        KARIGOR PRODUCTION
      </span>
    </a>
    <nav
      class="right-aligned"
    >
      <ul
        class="flex-row"
        id="nav-items"
      >
        <li
          class="mx-2"
        >
          <a
            href="#home"
          >
            Home
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            href="#about"
          >
            About
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            href="#contact"
          >
            Contact Us
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            href="#events"
          >
            Events
          </a>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Cultural
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Children Section
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Education
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Social Activities
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
